### PR TITLE
Adding export_climdex function

### LINKED ISSUE #16: Create function to export data in format for RClimDex
Function should export data for use in https://github.com/ECCC-CDAS/RClimDex

The format is specified in Appendix B here https://github.com/ECCC-CDAS/RClimDex/blob/master/inst/doc/manual.pdf

Two examples are attached 

[MN005050_Rclimdex.txt](https://github.com/IDEMSInternational/RInstatClimatic/files/7896593/MN005050_Rclimdex.txt)
[NP002051_Rclimdex.txt](https://github.com/IDEMSInternational/RInstatClimatic/files/7896595/NP002051_Rclimdex.txt)

### PR DESCRIPTION
**Pull request by lilyclements**
Fixes #16 

@dannyparsons this is ready to review

### CODE REVIEW

File: R/export_climdex.R [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+  assert_column_names(data, tmin)
+  if (!is.null(date)) assert_column_names(data, date)
+  if (!is.null(year)) assert_column_names(data, year)
+  if (!is.null(month)) assert_column_names(data, month)
+  if (!is.null(day)) assert_column_names(data, day)
+  checkmate::check_string(file)
+  file_type <- match.arg(file_type)
+  
+  if(is.null(year)) {
+    year <- ".year"
```
Comment (dannyparsons) [inline R/export_climdex.R:39]:
```suggestion
    year <- "year"
```

Comment (dannyparsons) [reply R/export_climdex.R:39]: I first had these as `.year` but then it doesn't look good in the output so I've tried to change to `year`

File: R/export_climdex.R [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    data[[year]] <- lubridate::year(data[[date]])
+  }
+  if(is.null(month)) {
+    month <- ".month"
+    data[[month]] <- lubridate::month(data[[date]])
+  }
+  if(is.null(day)) {
+    day <- ".day"
```
Comment (dannyparsons) [inline R/export_climdex.R:47]:
```suggestion
    day <- "day"
```

File: R/export_climdex.R [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    data[[day]] <- lubridate::day(data[[date]])
+  }
+  
+  file_path <- paste(file_path, file_type, sep = ".")
+  climdex_data <- data %>%
+    dplyr::select(c(.data[[year]], .data[[month]], .data[[day]], .data[[prcp]], .data[[tmax]], .data[[tmin]]))
+  
+  file_params <- utils::modifyList(list(x = climdex_data,
+                                        file = file_path,
+                                        row.names = FALSE),
```
Comment (dannyparsons) [inline R/export_climdex.R:57]: There should also be `col.names = FALSE` I think from looking at the example files.

**Comment by dannyparsons** [comment]
@lilyclements I've resolved the conflicts. Just make sure you pull before updating again.

File: R/export_climdex.R (hunk 3 above)
Comment (lilyclements) [reply R/export_climdex.R:57]: That’s the code I wanted! I was wondering why ‘row_names = FALSE’ wasn’t working. Cos they’re columns. Checks out.

Comment (dannyparsons) [reply R/export_climdex.R:57]: Well we also don't want the row names so it was half way there!

**Comment by lilyclements** [comment]
> If it's quick could this also be separated out into a prepare and export function?

This is now done and ready for re-review.

Note that `col.names = FALSE` does not work for .csv, so I'm using `write.table` always now, and added in `sep = ","` for if csv is selected [https://stackoverflow.com/questions/6750546/export-csv-without-col-names]

**Comment by dannyparsons** [comment]
Is that pushed here? I don't seem to see the prepare files.

**Comment by lilyclements** [comment]
There is `export_climdex` and `prepare_climdex` here. What other files were you expecting?

**Comment by dannyparsons** [comment]
Great thanks, sorry I think I might have just missed those before.

**Review by dannyparsons — APPROVED** [review]

**Pull request merged**